src/gwt/src/org/rstudio/core/client/MessageDisplay.java

```diff
@@ -196,6 +196,37 @@ public void showMessage(int type,
       .showModal();
    }
 
+   /**
+    * Show a messagebox with a clickable URL shown below the message text. The
+    * link will open in a new tab (e.g. target=_blank). 
+    *
+    * This mode is only supported for RStudio Server; on Desktop the link will not 
+    * be shown in the dialog.
+    */
+   public void showMessage(int type,
+                           String caption,
+                           String message,
+                           String urlLabel,
+                           String url)
+   {
+      createDialog(type, caption, message)
+         .addLink(urlLabel, url)
+         .showModal();
+   }
+
+   public void showMessage(int type,
+                           String caption,
+                           String message,
+                           String urlLabel,
+                           String url,
+                           Operation dismissed)
+   {
+      createDialog(type, caption, message)
+         .addLink(urlLabel, url)
```

Comment:
Is this link going to have target="_blank"?  We need that link to open up in a new tab or else the user will lose work in their session.  Also, I think the message should say: "Please try to login in a new tab and return here to continue the session".

Reply:
Yes, per the comment: "The link will open in a new tab (e.g. target=_blank)." That happens in MessageDialog.java (also in this review) via:

```java
Anchor anchor = new Anchor(label, url, "_blank");
```

Reply:
I'll modify the message to make it clearer that they should attempt to re-auth without closing everything out, since it should fix the RPC issues to do that.

Reply:
Re-worded to

> Unable to establish connection with session on {0}. Please click the link to try logging in again in a new tab, then return to resume your session.